astropy/modeling/tests/test_quantities.py

```diff
@@ -0,0 +1,232 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Tests specifically for models that use units and quantities."""
+
+from __future__ import (absolute_import, unicode_literals, division,
+                        print_function)
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+
+from ..core import Fittable1DModel, ModelDefinitionError, InputParameterError
+from ..parameters import Parameter, ParameterDefinitionError
+from ..models import Gaussian1D
+from ... import units as u
+from ...units import UnitsError, Quantity
+from ...tests.helper import pytest
+
+
+def test_quantities_as_parameters():
+    """
+    Basic tests for initializing general models (that do not require units)
+    with parameters that have units attached.
+    """
+
+    g = Gaussian1D(1 * u.J, 1 * u.m, 0.1 * u.m)
+    assert g.amplitude.value == 1.0
+    assert g.amplitude.unit is u.J
+    assert g.mean.value == 1.0
+    assert g.mean.unit is u.m
+    assert g.stddev.value == 0.1
+    assert g.stddev.unit is u.m
+
```

Comment:
Ah, well, the plan is have some kind of unit specs like `input_units` and `output_units` but for individual parameters as well (usually this is redundant, but if we want to force two parameters to have compatible units that would be the main use case).  I agree there should be some tests for that (even if they currently fail).